compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -451,7 +451,11 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] {
             // widening in `fourthTry` before adding to the constraint.
             if (frozenConstraint) isSubType(tp1, bounds(tp2).lo)
             else isSubTypeWhenFrozen(tp1, tp2)
-          alwaysTrue || {
+          alwaysTrue ||
+          frozenConstraint && (tp1 match {
```

Comment:
It would be good to have a test case that fails if the case is missing! Failing that, one can try to instrument the code to see whether the new logic makes a difference anywhere in the existing code base.

Reply:
@odersky the following code fails without this case:
```scala
val a = newTypeVar(TypeBounds.empty)
val b = newTypeVar(TypeBounds.empty)
a <:< b
assert(a frozen_<:< b)
```
I think that seems potentially problematic. I believe I actually ran across this issue half a year ago, when I was trying to make my exhaustivity checker use `Constraint` as a constraint solver. That being said, I have 0 idea on how to make TypeComparer enter this case w/o manually triggering it.